proto/p4/config/p4info.proto

```diff
@@ -134,44 +136,53 @@ message ActionProfile {
   int64 size = 4;  // max number of entries in action profile
 }
 
+// Corresponds to 'type' attribute for counter in P4 spec.
+enum CounterUnit {
+  UNSPECIFIED = 0;
+  BYTES = 1;
+  PACKETS = 2;
+  BOTH = 3;  // not part of the P4 spec yet but will be in the future
+}
+
 message Counter {
   Preamble preamble = 1;
-  // 0 (default value) if the counter is not direct, the id of the table to
-  // which the counter is attached otherwise
-  uint32 direct_table_id = 2;
-  // in the current P4 spec the attribute is called 'type', I believe 'unit' is
-  // more appropriate for this and I hope to see the spec updated
-  enum Unit {
-    UNSPECIFIED = 0;
-    BYTES = 1;
-    PACKETS = 2;
-    BOTH = 3;  // not part of the P4 spec yet but will be in the future
-  }
-  Unit unit = 3;
-  // number of entries in the counter array, for direct counters this is equal
-  // to the corresponding table's size
+  CounterUnit unit = 2;
+  // number of entries in the counter array
   int64 size = 4;
 }
 
+message DirectCounter {
+  Preamble preamble = 1;
+  // the id of the table to which the counter is attached
+  uint32 direct_table_id = 2;
+  CounterUnit unit = 3;
```

Comment:
should we try to keep the same "layout" as for `Counter`: `Preamble -> unit -> [direct_table_id | size ]`?
same applies to meters below